### PR TITLE
New feature Box component

### PR DESCRIPTION
**Pull request by srambach**
Implements very simple box component.

Note: Includes {{ styleModifier }} in the template for the time being as we figure out how it can be set up correctly.

Also fixes a typo in a couple of BoxShadow variables.

### CODE REVIEW

**Comment by srambach** [comment]
I've added the header, body, and footer sections.
Questions: 
- Does this need a stack layout inside? If so, it complicates the margin adjustments and is not as clean.
- In lieu of a "concept" variable, I've reused the header top margin and footer bottom margin to adjust the first/last child if those sections are missing. This still breaks the rule, but does not introduce a new type of variable and I've commented the reason for it in place.
@andresgalante @matthewcarleton @jgiardino

File: src/patternfly/components/Box/styles.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  --pf-c-box__footer--PaddingRight: var(--pf-global--spacer--xl);
+  --pf-c-box__footer--PaddingBottom: var(--pf-global--spacer--xl);
+  --pf-c-box__footer--PaddingLeft: var(--pf-global--spacer--xl);
+
+}
+
+.pf-c-box {
+  display: flex;
+  flex-direction: column;
+  height: 100%;
```
Comment (andresgalante) [inline src/patternfly/components/Box/styles.scss]: why do we set the height of the box?

Comment (srambach) [reply src/patternfly/components/Box/styles.scss]: If it's not set, then the box did not stretch (the body area) to fill the vertical space given, causing the footer to float up. If that's not the desired behavior, I can remove it.

Comment (jgiardino) [reply src/patternfly/components/Box/styles.scss]: When we place the box in a layout like grid or gallery, is the expectation that the box fill the height of it's container? Are there cases or layouts where that shouldn't happen?

File: src/patternfly/components/Box/examples/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const boxNoFooterExample = BoxNoFooterExample();
+  const boxNoHeaderExample = BoxNoHeaderExample();
+  const boxContentOnlyExample = BoxContentOnlyExample();
+
+  return (
+    <Documentation docs={Docs}>
+  <Example heading="Box">{boxBasicExample}</Example>
+  <Example heading="Box with no footer">{boxNoHeaderExample}</Example>
+  <Example heading="Box with no header">{boxNoFooterExample}</Example>
+  <Example heading="Bo with only a content section">{boxContentOnlyExample}</Example>
```
Comment (jgiardino) [inline src/patternfly/components/Box/examples/index.js]: This example heading has a typo.

**Comment by jgiardino** [comment]
LGTM!

File: src/patternfly/components/Box/styles.scss (hunk 1 above)
Comment (jgiardino) [reply src/patternfly/components/Box/styles.scss]: (@srambach's comment wasn't visible when I posted mine—I was replying to @andresgalante's original comment)

**Comment by andresgalante** [comment]
It needs a correct commit message to trigger the semantic release. Can I help you doing that?

**Comment by andresgalante** [comment]
@dgutride if I sqash and merge, it will squash up to the first commit message right?

**Comment by andresgalante** [comment]
@dgutride I'll create some issues for the technical debt in the box component

**Pull request merged**